src/lib/syncable_ledger/test_mask.ml

```diff
@@ -0,0 +1,108 @@
+open Core
+
+(* Input for testing the interaction between merkle *)
+
+module Make (Input : sig
+  val depth : int
+
+  val mask_layers : int
+end) =
+struct
+  open Merkle_ledger_tests.Test_stubs
+
+  module Hash = struct
+    type t = Hash.t [@@deriving sexp, hash, compare, bin_io, eq]
+
+    type account = Account.t
+
+    let merge = Hash.merge
+
+    let hash_account = Hash.hash_account
+
+    let to_hash = Fn.id
+
+    let empty_account = hash_account Account.empty
+  end
+
+  module Maskable_and_mask =
+    Merkle_ledger_tests.Test_mask.Make_maskable_and_mask_with_depth (Input)
+
+  module L = struct
+    open Merkle_ledger_tests.Test_stubs
+    module Base_db = Maskable_and_mask.Base_db
+    module Any_base = Maskable_and_mask.Any_base
+    module Base = Any_base.M
+    module Mask = Maskable_and_mask.Mask
+    module Maskable = Maskable_and_mask.Maskable
+    include Mask.Attached
+
+    (* Each account for a layer of a mask will all have the same balance.
+    Specifically, the top most layer will have a balance of `balance` and
+    every layer afterwards will have it's balance doubled *)
+    let load_ledger num_accounts (balance : int) : t * 'a =
+      let db = Base_db.create () in
+      let maskable = Any_base.cast (module Base_db) db in
+      let keys = Key.gen_keys num_accounts in
+      (* All the parents will have certain values *)
+      let initial_balance_multiplier = (1 lsl Input.mask_layers) * balance in
```

Comment:
For this test, I think readability is more important than efficiency. So probably better to write `Int.pow 2 Input.mask_layers` than to use a shift.